app/upload/views.py

```diff
@@ -26,6 +28,9 @@ def upload_document(service_id):
     file_content = request.files['document'].read()
 
     filename = request.form.get('filename')
+    file_extension = None
+    if filename:
+        file_extension = ''.join(pathlib.Path(filename).suffixes)[1:]
```

Comment:
```py
# Works most time
filename.split('.')[-1]
# Until this
"file.tar.gz".split('.')[-1]  # outputs gz

# outputs tar.gz
''.join(pathlib.Path("file.tar.gz").suffixes)[1:]
```

Reply:
Even `os.path.splitext` does not work properly with *.tar.gz* files. It feels like there should be a specialized library for proper file type/extension detection. 

I remember using *Apache Tika* in Java in the past, that was pretty good.

Reply:
Just noticed we have a function `get_mime_type`, will take a look at it later. 👍